Adding UUID generator

vmandrychenko opened the pull request:
- under IdGenerators

vmandrychenko committed:
Adding UUID generator

 - under IdGenerators

vmandrychenko committed:
Added docs for Id Generator

 - documented the example of uuid generator

fulrich reviewed (changes requested):
Requesting changes mostly to change the name to `default` to keep in line with the rest of the library.

vmandrychenko committed:
Improving IdGenerators DSL

 - using UUID as default, with default caller
 - corrected the name of defaultSize default caller implicit def

vmandrychenko committed:
Merge branch 'master' into uuid_generators
